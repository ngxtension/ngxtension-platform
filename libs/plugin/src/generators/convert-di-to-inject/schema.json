{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from constructor DI to inject function."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive/pipe you want to convert to use the inject function"
		},
		"includeReadonlyByDefault": {
			"type": "boolean",
			"aliases": ["includeReadonly", "includeReadonlyByDefault"],
			"description": "Whether to include readonly for injections by default. Default is false."
		},
		"useEsprivateFieldNotation": {
			"type": "boolean",
			"aliases": [
				"useESPrivateFieldNotation",
				"useEsprivateFieldNotation",
				"useEsPrivateFieldNotation"
			],
			"description": "Whether to replace TS private modifier with ES private field notation. Default is false."
		}
	}
}
