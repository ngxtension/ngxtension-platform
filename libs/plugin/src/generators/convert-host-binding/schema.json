{
	"$schema": "http://json-schema.org/schema",
	"$id": "ConvertHostBinding",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from host binding to host property."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to use the host property."
		}
	}
}
