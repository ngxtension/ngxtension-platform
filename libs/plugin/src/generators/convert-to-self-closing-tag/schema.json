{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to self closing tags."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to self closing tags"
		}
	}
}
