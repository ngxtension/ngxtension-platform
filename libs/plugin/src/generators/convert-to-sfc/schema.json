{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to single file components."
		},
		"path": {
			"type": "string",
			"description": "Path to the component to convert to a single file component."
		},
		"maxInlineTemplateLines": {
			"type": "number",
			"description": "If the template has more lines than this, it will be skipped.",
			"default": 200
		},
		"maxInlineStyleLines": {
			"type": "number",
			"description": "If the styles have more lines than this, they will be skipped.",
			"default": 200
		},
		"moveStyles": {
			"type": "boolean",
			"description": "Move styles to the single file component.",
			"default": false
		}
	}
}
