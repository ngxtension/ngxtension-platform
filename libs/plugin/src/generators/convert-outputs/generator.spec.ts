import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import convertOutputsGenerator from './generator';
import { ConvertOutputsGeneratorSchema } from './schema';

const filesMap = {
	notComponentNorDirective: `
import { Injectable } from '@angular/core';

@Injectable()
export class MyService {}
`,
	componentNoOutput: `
import { Component } from '@angular/core';

@Component({})
export class MyCmp {}
`,
	outputWithoutObservable: `
import { Component, Output, EventEmitter } from '@angular/core';

@Component({
  template: \` \`
})
export class MyCmp {
  @Output() outputWithoutType = new EventEmitter();
  @Output() normalOutput = new EventEmitter<string>();
}
`,
	component: `
import { Component, Output, EventEmitter } from '@angular/core';
import { toSignal } from '@angular/core/rxjs-interop';

@Component({
  template: \` \`
})
export class MyCmp {
  someObservable$ = of('test');

  @Output() outputWithoutType = new EventEmitter();

  @Output() private outputWithPrivateScope = new EventEmitter();
  @Output() protected outputWithProtectedScope = new EventEmitter()
  @Output() public outputWithPublicScope = new EventEmitter();

  @Output() normalOutput = new EventEmitter<string>();

  @Output() someOutput: EventEmitter<DataInterface> = new EventEmitter()

  @Output() normalOutput2: EventEmitter<string> = new EventEmitter<string>();

  @Output() outputFromSubject = new Subject();
  @Output() outputFromBehaviorSubject = new BehaviorSubject<number>();

  @Output() withObservable = this.someObservable$;
  @Output('withAlias') aliasOutput = new EventEmitter<string>();
  
  @Output() noInitializer: EventEmitter<string>;
  
  @Output() initializedInConstructor;
  
  constructor() {
    this.initializedInConstructor = new EventEmitter<string>();
  }

  ngOnInit() {
    let imABoolean = false;
    console.log(this.outputWithoutType);

    if (this.withTransform) {
      this.normalOutput.emit('test');
    }
  }

  handleClick() {
    if (true) {
      let test = this.outputWithoutType + this.normalOutput;
    }
  }
}
`,
} as const;

describe('convertOutputsGenerator', () => {
	let tree: Tree;
	const options: ConvertOutputsGeneratorSchema = {
		path: 'libs/my-file.ts',
	};

	function setup(file: keyof typeof filesMap) {
		tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
		tree.write('package.json', `{"dependencies": {"@angular/core": "17.3.0"}}`);
		tree.write(`libs/my-file.ts`, filesMap[file]);

		return () => {
			return [tree.read('libs/my-file.ts', 'utf8'), filesMap[file]];
		};
	}

	it('should not do anything if not component/directive', async () => {
		const readContent = setup('notComponentNorDirective');
		await convertOutputsGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should not do anything if no input', async () => {
		const readContent = setup('componentNoOutput');
		await convertOutputsGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should convert properly', async () => {
		const readContent = setup('component');
		await convertOutputsGenerator(tree, options);
		const [updated] = readContent();
		expect(updated).toMatchSnapshot();
	});

	it('should not add outputFromObservable import if not needed', async () => {
		const readContent = setup('outputWithoutObservable');
		await convertOutputsGenerator(tree, options);
		const [updated] = readContent();
		expect(updated).toMatchSnapshot();
	});
});
