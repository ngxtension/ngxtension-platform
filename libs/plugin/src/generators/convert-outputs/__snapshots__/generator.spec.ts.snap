// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertOutputsGenerator should convert properly 1`] = `
"
import { Component, output } from '@angular/core';
import { toSignal, outputFromObservable } from '@angular/core/rxjs-interop';

@Component({
  template: \` \`
})
export class MyCmp {
  someObservable$ = of('test');

  outputWithoutType = output();

  protected outputWithPrivateScope = output();
  protected outputWithProtectedScope = output();
  public outputWithPublicScope = output();

  normalOutput = output<string>();

  someOutput = output<DataInterface>();

  normalOutput2 = output<string>();

  outputFromSubject = new Subject();
  outputFromBehaviorSubject = new BehaviorSubject<number>();

  withObservable = outputFromObservable(this.someObservable$);
  aliasOutput = output<string>({ alias: 'withAlias' });
  
  noInitializer = output<string>();
  
  initializedInConstructor = output<string>();
  
  constructor() {
  }

  ngOnInit() {
    let imABoolean = false;
    console.log(this.outputWithoutType);

    if (this.withTransform) {
      this.normalOutput.emit('test');
    }
  }

  handleClick() {
    if (true) {
      let test = this.outputWithoutType + this.normalOutput;
    }
  }

    /** TODO(migration): you may want to convert this to a normal output */
    _outputFromSubject = outputFromObservable(this.outputFromSubject, { alias: 'outputFromSubject' });
    /** TODO(migration): you may want to convert this to a normal output */
    _outputFromBehaviorSubject = outputFromObservable(this.outputFromBehaviorSubject, { alias: 'outputFromBehaviorSubject' });
}
"
`;

exports[`convertOutputsGenerator should not add outputFromObservable import if not needed 1`] = `
"
import { Component, output } from '@angular/core';

@Component({
  template: \` \`
})
export class MyCmp {
  outputWithoutType = output();
  normalOutput = output<string>();
}
"
`;
